flask_app.py

```diff
@@ -0,0 +1,155 @@
+
+    Returns:
+        str: The rendered HTML page or a JSON response with the UID and HTTP status code 200.
+    """
+    if request.method == 'POST':
+        if 'file' not in request.files:
+            flash('No file part')
+            return redirect(request.url)
+        file = request.files.get('file')
+        if file.filename == '':
+            flash('No selected file')
+            return redirect(request.url)
+        email = request.form.get('email')
+        if email:
+            uid = save_upload_with_user(file, email)
+        else:
+            uid = save_upload(file)
+        return jsonify({'uid': uid}), 200
+    return render_template("upload.html")
+
+
+@app.route('/status/<uid>')
+def status(uid):
+    """
+    Retrieves the processing status of the file associated with the given UID.
+    If the file is processed, it loads the output and generates a JSON response
+    with the processing status, filename, timestamp, and output.
+    If the file is not found or not processed, it generates a JSON response
+    indicating the status as 'not found'.
+
+    Args:
+        uid (str): The UID of the file.
+
+    Returns:
+        Response: A JSON response with the processing status and information.
+    """
+    with Session() as session:
+        file_data = session.query(Upload).filter_by(uid=uid).first()
+        if file_data:
+            if file_data.status == status_done:
+                output = load_output(f"{uid}.json")
+                status_info = save_to_json(uid, file_data.status, file_data.filename, file_data.finish_time, output)
+            else:
+                status_info = save_to_json(uid, file_data.status, file_data.filename, file_data.finish_time)
+            return Response(json.dumps(status_info), mimetype='application/json')
+    return jsonify({'status': 'not found'}), 404
+
+
+@app.route('/search', methods=['POST', 'GET'])
+def search():
+    """
+    Handles the search functionality. Accepts both POST and GET requests.
+    If a POST request is received with a non-empty UID, it redirects to the
+    'status' route with the UID as a parameter.
+    If a POST request is received with an empty UID, it flashes a message
+    indicating that the UID should be entered and redirects back to the search page.
+    If a POST request is received with both email and filename provided, it searches
+    for the latest upload with that email and filename and redirects to 'status' route
+    with the UID of the latest upload.
+    If a GET request is received, it renders the 'search.html' template.
+
+    Returns:
+        str: The rendered HTML page or a redirect response.
+    """
+    if request.method == 'POST':
+        uid = request.form.get('uid')
+        email = request.form.get('email')
+        filename = request.form.get('filename')
+        if uid:
+            return redirect(url_for('status', uid=uid))
+        elif email and filename:  # Only search if both email and filename are provided
```

Comment:
you should have redirected this too

Reply:
I did redirected this, but first I found the uid by searching email + filename in the database